### DISCUSSION TITLE
Animated layout doesn't work with FlatList

### DISCUSSION THREAD

**Discussion by Pietro-Putelli** [software-mansion/react-native-reanimated#2738] [Q&A]
Reanimated version: 2.3.0
When I try to use FlatList instead of ScrollView, as shown in the tutorial below, the Layout.springify() doesn't work.
https://docs.swmansion.com/react-native-reanimated/docs/tutorials/LayoutAnimations/layoutAnimations

Someone suggests to apply this path: https://patch-diff.githubusercontent.com/raw/piaskowyk/react-native/pull/1.patch.
How can I apply the patch?
Is there an alternative solution?

**Comment by piaskowyk** [comment]
Hey, Try to use it - https://github.com/software-mansion/react-native-reanimated/pull/2674
```js
<Animated.FlatList // use Animated.FlatList instead of FlatList
  itemLayoutAnimation={Layout.springify()} // this animation will be apply as layout property for every list item
  data={itemsData} // as always
  renderItem={renderItem} // as always
/>
```

**Reply by Pietro-Putelli** [reply]
Hi Krzysztof thank you really much for your answer, but it doesn't work with react-native-reanimated ^2.3.0.
Here's the code:

```
<AnimatedFlatList
          inverted
          data={messages}
          itemLayoutAnimation={Layout.springify()}
          renderItem={({ item }) => {
            return (
              <Animated.View
                entering={LightSpeedInLeft}
                exiting={LightSpeedOutRight}
                key={item.id}
                style={{...}}
              ></Animated.View>
            );
          }}
          keyExtractor={(item) => item.id}
        />
```
